packages/domains-table/src/domains-table/domains-table-ssl-cell.tsx

```diff
@@ -4,24 +4,49 @@ import { lock } from '@wordpress/icons';
 import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
 
-interface DomainsTableSSLCellProps {
+interface DomainsTableSslCellProps {
 	domainManagementLink: string;
 	sslStatus: DomainData[ 'ssl_status' ];
+	hasWpcomManagedSslCert?: DomainData[ 'has_wpcom_managed_ssl_cert' ];
 }
 
-export default function DomainsTableSSLCell( {
+export default function DomainsTableSslCell( {
 	domainManagementLink,
 	sslStatus,
-}: DomainsTableSSLCellProps ) {
+	hasWpcomManagedSslCert,
+}: DomainsTableSslCellProps ) {
 	const translate = useTranslate();
+	// WordPress.com managed subdomains (e.g. *.wordpress.com, *.wpcomstaging.com, etc.)
+	// are covered by a wildcard SSL cert, so we display them as 'Active'.
+	const isActiveSsl = sslStatus === 'active' || hasWpcomManagedSslCert;
+	const isPendingSsl = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const domainHasSsl = sslStatus || hasWpcomManagedSslCert;
 
-	const isPendingSSL = sslStatus === 'pending' || sslStatus === 'newly_registered';
+	const Element = ( { children, ...props }: { children: React.ReactNode; className: string } ) => {
+		if ( sslStatus ) {
+			return (
+				<a
+					href={ `${ domainManagementLink }?ssl-open=true` }
+					onClick={ ( event ) => event.stopPropagation() }
+					{ ...props }
+				>
+					{ children }
+				</a>
+			);
+		}
+
+		if ( hasWpcomManagedSslCert ) {
+			return <span { ...props }>{ children }</span>;
+		}
 
-	const getSSLStatusText = () => {
-		if ( sslStatus === 'active' ) {
+		return '-';
+	};
```

Comment:
Let's avoid defining a React component inside another component and either move `Element` to the topmost scope (which would necessitate some more prop passing, but could be OK) or do something similar to the previous implementation, i.e., along these lines:

```tsx
	const buttonClassNames = clsx( 'domains-table-row__ssl-status-button', {
		[ 'domains-table-row__ssl-status-button__active' ]: isActiveSsl,
		[ 'domains-table-row__ssl-status-button__pending' ]: isPendingSsl,
		[ 'domains-table-row__ssl-status-button__disabled' ]: sslStatus === 'disabled',
	} );
	let button: React.ReactElement | string;

	if ( sslStatus ) {
		button = (
			<a
				className={ buttonClassNames }
				href={ `${ domainManagementLink }?ssl-open=true` }
				onClick={ ( event ) => event.stopPropagation() }
			>
				{ getSslStatusText() }
			</a>
		);
	} else if ( hasWpcomManagedSslCert ) {
		button = <span className={ buttonClassNames }>{ getSslStatusText() }</span>;
	} else {
		button = '-';
	}
```

FYI, defining a React component within another component isn't strictly forbidden, but it is an anti-pattern. The clearest argument against it is that it decreases performance (the inner component is redefined on every render, meaning React can't properly optimize re-renders). Performance isn't a huge concern here, it's more that I don't think there's anything really compelling us to go against the convention here.

Reply:
Good point, thanks! I feel like I've seen this pattern before but hadn't connected to dots on the poor optimization impact, thank you for elaborating!